full-service/src/json_rpc/v1/api/wallet.rs

```diff
@@ -741,14 +741,39 @@ where
             .map_err(format_error)?,
         },
         JsonCommandRequest::get_transaction_log { transaction_log_id } => {
-            let (transaction_log, associated_txos, _) = service
-                .get_transaction_log(&transaction_log_id)
-                .map_err(format_error)?;
-
-            let json_tx_log = json_rpc::v1::models::transaction_log::TransactionLog::new(
-                &transaction_log,
-                &associated_txos,
-            );
+            // Check whether the transaction_log_id actually refers to the txo_id of a
+            // received transaction.
+            let txo_id = TxoID(transaction_log_id.clone());
+            let json_tx_log = if let Ok((txo, _)) = service.get_txo(&txo_id) {
+                // A txo was found, determine which address it was received at, if any.
+                let subaddress_b58 = if let Some(subaddress_index) = txo.subaddress_index {
+                    if let Some(ref account_id) = txo.account_id {
+                        let account_id = AccountID(account_id.clone());
+                        service.get_address_for_account(
+                            &account_id,
+                            subaddress_index,
+                        )
+                        .map(|assigned_sub| assigned_sub.public_address_b58)
+                        .ok()
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
```

Comment:
This can be cleaned up by using a single match statement

```suggestion
                let subaddress_b58 = match (txo.subaddress_index, txo.account_id) {
                    (Some(subaddress_index), Some(account_id)) => {
                        let account_id = AccountID(account_id);
                        service
                            .get_address_for_account(&account_id, subaddress_index)
                            .map(|assigned_sub| assigned_sub.public_address_b58)
                            .ok()
                    }
                    _ => None,
                };
```

Reply:
Nice improvement.